code/modules/projectiles/projectile.dm

```diff
@@ -1131,3 +1131,19 @@
 
 #undef MOVES_HITSCAN
 #undef MUZZLE_EFFECT_PIXEL_INCREMENT
+
+/// Fire a projectile from this atom at another atom
+/atom/proc/fire_projectile(projectile_type, atom/target, sound, firer)
+	if (!isnull(sound))
+		playsound(src, sound, vol = 100, vary = TRUE)
+
+	var/turf/startloc = get_turf(src)
+	var/obj/projectile/bullet = new projectile_type(startloc)
+	bullet.starting = startloc
+	bullet.firer = firer || src
```

Comment:
```suggestion
	bullet.firer ||= src
```
...Will firer ever be set / non-null in projectile New?

Reply:
`firer` is an optional argument passed into the proc, rather than it being `bullet.firer = bullet.firer || src`
I can rename it if that is making it unclear